Add doNotAddAttributeQuotes setting to disable automatic quotes

ssigwart opened the pull request:
<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `main` branch.
* Include a description of the proposed changes and how to test them.
-->

This depends on microsoft/vscode-html-languageservice#112

This allows users to disable the automatic quotes after an HTML attribute.  I'm not sure the best way to test this since it depends on the pull request in `microsoft/vscode-html-languageservice`.  I tested by manually modifying `extensions/html-language-features/server/_modules/vscode-html-languageservice/lib/umd/services/htmlCompletion.js` to apply the changes.

A simple test is to use following and trigger the `class` completion:
```html
<div clas|
```
With this setting enabled, it should be `class=|`, not `class="|"`.

aeschli commented on extensions/html-language-features/package.json:
```diff
@@ -188,6 +188,12 @@
           "default": true,
           "description": "%html.autoClosingTags%"
         },
+        "html.doNotAddAttributeQuotes": {
```
this should follow the style of the existing settings, e.g. html.completion.attributeQuotes

aeschli commented:
I'd rather avoid new settings as they make the product more complex and add are costly to maintain.
So before adding a setting I'd like to see more user request for it.

ssigwart commented:
Thanks, @aeschli.  Where would other users request it?  Should I add an issue so it can go into the backlog candidate flow.

Also, do you think I'd be able to write an extension to accomplish this?  I can't think of a way without writing a whole LSP extension for HTML, which would be overkill.

For reference, I personally prefer typing the quote myself.  I can probably train myself to stop type the quote, but the bigger issue is that is seems like VSCode doesn't recognize trigger characters if they were part of a completion.  That seems to be the issue in https://github.com/zignd/HTML-CSS-Class-Completion/issues/235 and https://github.com/gencer/SCSS-Everywhere/issues/36 for CSS completion.  Currently, the way to work around that is to let `class="|"` auto-complete, then hit backspace, then type the quote and CSS completion will work.  If the quotes weren't auto-inserted, it would be more intuitive.

Alternatively, would it be acceptable to write a patch where if you have `editor.autoClosingOvertype` set to `never` and type `"` in `class="|"` just after it's inserted, it will assume you meant to type the opening brace an overwrite it similar to the `editor.autoClosingOvertype` setting?  Currently, it winds up as `class=""|"`.

aeschli commented:
Yes, an issue that goes through backlog candidate would be the way to get other users feedback/

I would be more open to it we could make setting a bit more general. Some users might prefer single quotes 
`html.completion.attributeDefaultValue=empty | singleQuotes | doubleQuotes`

ssigwart commented:
Thanks, @aeschli.  I like that idea.  I updated this pull request and microsoft/vscode-html-languageservice#112 to do that.  I also opened #131144 to track if there's interest in this.

ssigwart commented:
@aeschli, I updated this to match the changes in https://github.com/microsoft/vscode-html-languageservice/pull/112.  I rebased it and now there's an error with typescript, which I didn't touch, so I ignored it.  Please let me know if I should rebased at a later point once that error is resolved.

aeschli commented:
That looks good, thanks @ssigwart ! I'll merge it next week.

ssigwart commented:
Thanks, @aeschli.  I'm really looking forward to this setting.

aeschli commented:
Thanks @ssigwart !
